playground/house.py

```diff
@@ -45,9 +45,31 @@ def play(self):
 
         self.board.render_figure()
 
+    def simulation(self):
+        while True:
+            turn = self.referee.turn_check(self.board.log)
+            if self.player_first == (turn == "b"):
+                order = "player"
+                stone = Bot("b").tei_bot(self.board.log)
+            else:
+                order = "bot"
+                stone = Bot("w").tei_bot(self.board.log)
+
+            if self.referee.valid_check(stone, self.board.log):
+                self.board.put_stone(stone)
+                print(f"{order} stone: {stone.x},{stone.y},{turn}")
+                placement_result = self.referee.end_check(self.board.log)
+                tie_check = self.referee.tie_check(self.board.log, self.board)
+                if (placement_result != "keep play") | tie_check:
+                    break
+            else:
+                print("invalid stone input")
+
+        self.board.render_figure()
```

Comment:
does it also have to render figure, if the stone input is invalid? Please check @ths2o @hyoungseok

Reply:
Yes, you are right!

Reply:
and this is hip